File: application/router/Router.php
<?php

require_once '..\application\db\DB.php';
require_once '..\application\game\Game.php';
require_once '..\application\user\User.php';

class Router {

    private $user;
    private $game;
    private $db;

    public function __construct() {
        $this->db = new DB();
        $this->game = new Game($this->db);
        $this->user = new User($this->db);
    }

    private function bad($text) {
        return array(
            'result' => false,
            'error' => $text
        );
    }
    private function good($data) {
        return array(
            'result' => true,
            'data' => $data
        );
    }

    private function login($params) {
        $token = $this->user->login((object) $params);
        return ($token) ?
            $this->good($token) :
            $this->bad('authorization fail');
    }

    public function answer($params) {
        $method = $params['method'];
        if ($method) {
            unset($params['method']);
            // вызвать методы АПИ
            switch ($method) {
                case 'login': return $this->login($params); break;
                //...
            }
            // вызвать команду игры
            $commands = $this->game->getCommands();
            if ($method == $commands->GET_STRUCT) {
                return $this->good($this->game->getStruct());
            }
            foreach ($commands as $command) {
                if ($command === $method) {
                    $result = $this->game->executeCommand($method, (object) $params);
                    return ($result) ?
                        $this->good($this->game->getStruct()) :
                        $this->bad('game method return false');
                }
            }
            // вызвать другие команды
            //...
        }
        return $this->bad('method does not exist');
    }
}

File: SmartHouse/application/router/Router.py
import random
'''
from application.game.Game import Game
from application.db.DB import DB
from application.user.UserManager import UserManager
'''

class Router:
    def __init__(self, app, web, mediator):
        self.web = web
        self.mediator = mediator
        
        routers = [
            ('GET', '/api/registration/{login}/{hash}', self.registrationHandler),
            ('GET', '/api/login/{login}/{hash}', self.loginHandler),
            ('GET', '/api/logout/{token}', self.logoutHandler),
            ('*', '/', self.staticHandler)
        ]
        app.router.add_static('/js/', path=str('./public/js/'))
        for route in routers:
            app.router.add_route(route[0], route[1], route[2])

    def staticHandler(self, request):
        return self.web.FileResponse('./public/index.html')

    """ Войти в систему """
    def loginHandler(self, request):
        login = request.match_info.get('login')
        hash = request.match_info.get('hash')
        result = self.mediator.get(self.mediator.getTriggerTypes()['LOGIN'], { 'login': login, 'hash': hash })
        return self.answer(result, 'Вы не авторизовались!')
    
    """ Выйти из системы """
    def logoutHandler(self, request):
        token = request.match_info.get('token')
        result = self.mediator.get(self.mediator.getTriggerTypes()['LOGOUT'], { 'token': token })
        return self.answer(result, 'Ошибка деавторизации!')

    """ Регистрация в системе """
    def registrationHandler(self, request):
        login = request.match_info.get('login')
        hash = request.match_info.get('hash')
        result = self.mediator.get(self.mediator.getTriggerTypes()['REGISTRATION'], { 'login': login, 'hash': hash })
        return self.answer(result, 'Вы не зарегистрировались!')

    def answer(self, data, error):
        if data:
            return self.good(data)
        else:
            return self.bad(error)

    def good(self, data):
        return self.web.json_response({
            'result': 'ok',
            'data': data
        })

    def bad(self, error):
        return self.web.json_response({
            'result': 'error',
            'data': error
        })